components/segmenter/src/sentence.rs

```diff
@@ -121,32 +123,46 @@ impl SentenceSegmenter {
             payload: DataPayload::from_static_ref(
                 crate::provider::Baked::SINGLETON_SENTENCE_BREAK_DATA_V2_MARKER,
             ),
+            payload_locale_override: None,
         }
     }
 
-    icu_provider::gen_any_buffer_data_constructors!(() -> error: DataError,
-        functions: [
-            new: skip,
-            try_new_with_any_provider,
-            try_new_with_buffer_provider,
-            try_new_unstable,
-            Self,
-        ]
+    icu_provider::gen_any_buffer_data_constructors!((locale) -> error: DataError,
+        /// Constructs a [`SentenceSegmenter`] for a given locale and using compiled data.
+        ///
+        /// ✨ *Enabled with the `compiled_data` Cargo feature.*
+        ///
+        /// [📚 Help choosing a constructor](icu_provider::constructors)
     );
 
     #[doc = icu_provider::gen_any_buffer_unstable_docs!(UNSTABLE, Self::new)]
-    pub fn try_new_unstable<D>(provider: &D) -> Result<Self, DataError>
+    pub fn try_new_unstable<D>(provider: &D, locale: &DataLocale) -> Result<Self, DataError>
```

Comment:
Nit: This function signature is still not what we agreed in the issue, but it can be fixed later (but before 2.0)